src/LodeRunner.Data/README.md

```diff
@@ -2,13 +2,94 @@
+|4    | Create secret file    | `echo $COSMOS_RW_KEY > ./src/LodeRunner.API/secrets/CosmosKey` |
+
+The following block has all of the environment variable assignments in one place to copy and paste:
+
+```bash
+export CosmosDB_RG=rg-ngsa-asb-dev-cosmos
+export CosmosDB_ACCT=ngsa-asb-dev-cosmos
+export LR_DB=LodeRunnerDB
+export TEST_DB=LodeRunnerTestDB
+export LR_COL=LodeRunner
+export CosmosDB_URL=https://ngsa-asb-dev-cosmos.documents.azure.com:443/
+export COSMOS_RW_KEY=$(eval az cosmosdb keys list -n $CosmosDB_ACCT -g $CosmosDB_RG --query primaryMasterKey -o tsv)
+echo $COSMOS_RW_KEY > ./src/LodeRunner.API/secrets/CosmosKey
+```
+
+## CosmosDB Firewall IP Ranges
+
+Certain loderunner applications (i.e. LodeRunner, LodeRunner.API) add your IP address to allow access to CosmosDB.  If you run the app and cannot access the CosmosDB due to firewall you may see the error output:
+
+> `"ExceptionMessage": "Repository test for LodeRunnerDB:LodeRunner failed."
+
+To solve this we need allow your IP through the firewall.
+
+### Allowing IP Access via the Portal
+
+1. Navigate to the Azure Portal
+2. Go to the associated `Azure Cosmos DB account`
+3. Under `Settings`, go to `Firewall and virtual networks`
+4. Add your IP address under `Firewall` > `IP`
+5. Save
+6. A notification with "Updating Firewall configuration" should appear at the top
+7. It will take a little time to update. Once the firewall is updated you will be able to run the app.
+
+### Allowing IP Access via CLI
+<!-- markdownlint-disable MD033 -->
+You will need to have set the environment variables from the previous section [CosmosDB Secrets](#cosmosdb-secrets).  The following table has the values that you are setting and the command.
+
+|Step | Description        | Command |
+| --- |:-----------        |:------- |
+|   1  | **YOUR_IP**           |  `export YOUR_IP=$(eval dig +short myip.opendns.com @resolver1.opendns.com)` |
+|   2  | **IPS_FOR_FIREWALL**  | `export IPS_FOR_FIREWALL=az cosmosdb show -g $CosmosDB_RG -n $CosmosDB_ACCT | jq -r '.ipRules | map(.ipAddressOrRange) | @csv' | tr -d '"'` |
+|   3  | Check if your IP already allowed. <br> **Skip to step 6 if your IP is in the list** highlighted in <font color=red>red</font> by default. If you **do not see your address** then you will need to add it by proceeding to step 5.           | `grep "$YOUR_IP" <<< "$IPS_FOR_FIREWALL"` |
+|  4 | Create **ALL_IPS** list | `export ALL_IPS="\"${IPS_FOR_FIREWALL},${YOUR_IP}\""` |
+|  5    | Update the allowed IPs for CosmosDB | `az cosmosdb update -n $CosmosDB_ACCT -g $CosmosDB_RG --ip-range-filter $ALL_IPS` |
+
+The following block is all of the shell commands pulled together with a conditional to check if you're IP is already allowed.  If not it will add it.
+
+```bash
+export YOUR_IP=$(eval dig +short myip.opendns.com @resolver1.opendns.com)
+export IPS_FOR_FIREWALL=az cosmosdb show -g $CosmosDB_RG -n $CosmosDB_ACCT | jq -r '.ipRules | map(.ipAddressOrRange) | @csv' | tr -d '"'
```

Comment:
To execute successfully, I had to wrap the command in ``:
```bash
export IPS_FOR_FIREWALL=`az cosmosdb show -g $CosmosDB_RG -n $CosmosDB_ACCT | jq -r '.ipRules | map(.ipAddressOrRange) | @csv' | tr -d '"'`
```

Reply:
It should have been:
```bash
export IPS_FOR_FIREWALL=$(eval az cosmosdb show -g $CosmosDB_RG -n $CosmosDB_ACCT | jq -r '.ipRules | map(.ipAddressOrRange) | @csv' | tr -d '"')
```
I tested it and updated the PR.